crates/zizmor/src/audit/use_trusted_publishing.rs

```diff
@@ -201,6 +201,77 @@ audit_meta!(
 );
 
 impl UseTrustedPublishing {
+    // Detect whether this job is using crates.io Trusted Publishing by checking for
+    // a prior step that uses rust-lang/crates-io-auth-action and that this step
+    // wires its token into cargo via env or inline token usage.
+    fn uses_crates_io_trusted_publishing(
+        &self,
+        step: &crate::models::workflow::Step,
+        run: &str,
+    ) -> bool {
+        // Collect IDs of prior steps in the same job that use rust-lang/crates-io-auth-action
+        let mut auth_step_ids: Vec<&str> = Vec::new();
+
+        for prior in step.job().steps() {
+            if prior.index >= step.index {
+                break;
+            }
+
+            // Only consider uses: steps that reference the crates-io auth action
+            let Some(common_uses) = prior.uses() else { continue };
+            let github_actions_models::common::Uses::Repository(repo_uses) = common_uses else {
+                continue;
+            };
+
+            if repo_uses.owner.eq_ignore_ascii_case("rust-lang")
+                && repo_uses.repo.eq_ignore_ascii_case("crates-io-auth-action")
+            {
+                if let Some(id) = &prior.id {
+                    auth_step_ids.push(id);
+                }
+            }
+        }
+
+        if auth_step_ids.is_empty() {
+            return false;
+        }
+
+        // Helper: does a string reference any of the known auth step outputs?
+        let references_auth_output = |s: &str| {
+            auth_step_ids
+                .iter()
+                .any(|id| s.contains(&format!("steps.{id}.outputs.token")))
```

Comment:
Similarly: could you write this as a `ContextPattern`? That'll ensure we match the various syntax variants context accesses.

(It also implements `FromStr`)